godot-codegen/src/lib.rs

```diff
@@ -33,44 +34,68 @@ use proc_macro2::{Ident, TokenStream};
 use quote::{quote, ToTokens};
 use std::path::{Path, PathBuf};
 
+pub type SubmitFn = dyn FnMut(PathBuf, TokenStream);
+
+fn write_file(path: &Path, contents: String) {
+    let dir = path.parent().unwrap();
+    let _ = std::fs::create_dir_all(dir);
+
+    std::fs::write(path, contents)
+        .unwrap_or_else(|e| panic!("failed to write code file to {};\n\t{}", path.display(), e));
+}
+
 pub fn generate_sys_files(
     sys_gen_path: &Path,
     h_path: &Path,
     watch: &mut godot_bindings::StopWatch,
 ) {
-    let mut out_files = vec![];
+    #[cfg(not(feature = "codegen-fmt"))]
+    let mut submit_fn = |path: PathBuf, tokens: TokenStream| {
+        write_file(&path, format_tokens(tokens));
+    };
+
+    #[cfg(feature = "codegen-fmt")]
+    let mut submit_fn = |path, tokens| {
+        write_file(&path, tokens.to_string());
+    };
```

Comment:
Also, the two conditionals are swapped 🙂 

If feature `codegen-fmt` is enabled, then you should run `format_tokens`, otherwise not.
(I got an internal compiler error when swapping and incrementally compile further; had to clean).

If formatting turns out to have negligible performance impact, we can always enable it; but I'd suggest to do this outside of this PR.

Reply:
Oops, good catch, I put it there for testing 🙈 Corrected that! For enabling it by default, somebody should probably do proper measurements! For me it seems like it added about 200-400 ms in total.